pkg/generator/ingress_translator_test.go

```diff
@@ -1273,7 +1282,283 @@ func TestIngressTranslatorWithUpstreamTLS(t *testing.T) {
+										Exact: "foo",
+									},
+								},
+							},
+						}},
+						"/test",
+						[]*route.WeightedCluster_ClusterWeight{
+							envoy.NewWeightedCluster("servicens/servicename", 100, map[string]string{"baz": "gna"}),
+						},
+						0,
+						map[string]string{"foo": "bar"},
+						""),
+					},
+				),
+			}
+
+			return &translatedIngress{
+				name: types.NamespacedName{
+					Namespace: "simplens",
+					Name:      "simplename",
+				},
+				externalSNIMatches: []*envoy.SNIMatch{},
+				localSNIMatches:    []*envoy.SNIMatch{},
+				clusters: []*v3.Cluster{
+					envoy.NewCluster(
+						"servicens/servicename",
+						5*time.Second,
+						lbHTTPSEndpoints,
+						false,
+						&envoycorev3.TransportSocket{
+							Name:       wellknown.TransportSocketTls,
+							ConfigType: typedConfig(false, configmapCert),
+						},
+						v3.Cluster_STATIC,
+					),
+				},
+				externalVirtualHosts:    vHosts,
+				externalTLSVirtualHosts: []*route.VirtualHost{},
+				localVirtualHosts:       vHosts,
+				localTLSVirtualHosts:    []*route.VirtualHost{},
+			}
+		}(),
+	}, {
+		name: "invalid CA from configmap",
+		in: ing("simplens", "simplename", func(ing *v1alpha1.Ingress) {
+			ing.Spec.Rules[0].HTTP.Paths[0].RewriteHost = ""
+			ing.Spec.Rules[0].HTTP.Paths[0].Splits[0].IngressBackend.ServicePort = intstr.FromInt(443)
+		}),
+		state: []runtime.Object{
+			svc("servicens", "servicename"),
+			eps("servicens", "servicename"),
+			func() runtime.Object {
+				s := caSecret.DeepCopy()
+				delete(s.Data, certificates.CaCertName)
+				return s
+			}(),
+			invalidCAConfigmap,
+		},
+		want: func() *translatedIngress {
+			vHosts := []*route.VirtualHost{
+				envoy.NewVirtualHost(
+					"(simplens/simplename).Rules[0]",
+					[]string{"foo.example.com", "foo.example.com:*"},
+					[]*route.Route{envoy.NewRoute(
+						"(simplens/simplename).Rules[0].Paths[/test]",
+						[]*route.HeaderMatcher{{
+							Name: "testheader",
+							HeaderMatchSpecifier: &route.HeaderMatcher_StringMatch{
+								StringMatch: &envoymatcherv3.StringMatcher{
+									MatchPattern: &envoymatcherv3.StringMatcher_Exact{
+										Exact: "foo",
+									},
+								},
+							},
+						}},
+						"/test",
+						[]*route.WeightedCluster_ClusterWeight{
+							envoy.NewWeightedCluster("servicens/servicename", 100, map[string]string{"baz": "gna"}),
+						},
+						0,
+						map[string]string{"foo": "bar"},
+						""),
+					},
+				),
+			}
+
+			return &translatedIngress{
+				name: types.NamespacedName{
+					Namespace: "simplens",
+					Name:      "simplename",
+				},
+				externalSNIMatches: []*envoy.SNIMatch{},
+				localSNIMatches:    []*envoy.SNIMatch{},
+				clusters: []*v3.Cluster{
+					envoy.NewCluster(
+						"servicens/servicename",
+						5*time.Second,
+						lbHTTPSEndpoints,
+						false,
+						&envoycorev3.TransportSocket{
+							Name:       wellknown.TransportSocketTls,
+							ConfigType: typedConfig(false, nil),
```

Comment:
Passing nil here (also tested with a missing cm) means that we end up having no validation since this is passed to https://github.com/knative-extensions/net-kourier/blob/2a72a55e82ed530aca30a81286ad1b7e6f321761/pkg/generator/ingress_translator_test.go#L2078.
Check comments here too: https://github.com/knative-extensions/net-kourier/blob/2a72a55e82ed530aca30a81286ad1b7e6f321761/vendor/github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3/common.pb.go#L816-L835
Would that allow tls communication without validation or connection is failing?
Also do you think that this affects us https://github.com/envoyproxy/envoy/issues/19326 downstream? :thinking:

Reply:
What about this part:

" Note 
 // that if a CRL is provided for any certificate authority in a trust chain, a CRL must be 
 // provided for all certificate authorities in that chain. Failure to do so will result in 
 // verification failure for both revoked and unrevoked certificates from that chain."